src/ruby.ts

```diff
@@ -0,0 +1,83 @@
+import { exec } from "child_process";
+import { promisify } from "util";
+import * as fs from "fs";
+
+import * as vscode from "vscode";
+
+const asyncExec = promisify(exec);
+const asyncReadFile = promisify(fs.readFile);
+
+export class Ruby {
+  private workingFolder: string;
+  private managerConfiguration: { [key: string]: string };
+
+  constructor() {
+    this.workingFolder = vscode.workspace.workspaceFolders![0].uri.fsPath;
+
+    this.managerConfiguration = vscode.workspace
+      .getConfiguration("rubyLsp")
+      .get("rubyVersionManager")!;
+  }
+
+  async activateRuby() {
+    switch (this.managerConfiguration.manager) {
+      case "chruby":
+        await this.activateChruby();
+        break;
+      default:
+        this.activateShadowenv();
+        break;
+    }
+  }
+
+  activateShadowenv() {
+    vscode.extensions.getExtension("shopify.vscode-shadowenv")?.activate();
+  }
+
+  async activateChruby() {
+    if (!fs.existsSync(`${this.workingFolder}/.ruby-version`)) {
+      vscode.window.showErrorMessage(
+        "Attempted to activate chruby environment, but no .ruby-version file was found."
+      );
+      return;
+    }
+
+    if (!fs.existsSync(this.managerConfiguration.path)) {
+      vscode.window.showErrorMessage(
+        `Attempted to activate chruby environment, but the path provided does not exist (${this.managerConfiguration.path}).`
+      );
+      return;
+    }
```

Comment:
Can we not make the `path` optional and expect to find it on the default path? I imagine that if someone is using `chruby` to manage their Ruby versions, they would have it properly setup and running `chruby <version>` would just work.

Reply:
Chruby is installed in different paths on Linux and macos, so we'd need to look it up based on the architecture.

Also, I'm not sure the nodejs will inherit `chruby`, since it's not an executable. It's a bash function defined when you source the `chruby.sh` script. You think we don't need to source it and it should just work?